netcore/sample/Makefile

```diff
@@ -0,0 +1,24 @@
+all: run
+
+URL:=https://dotnetcli.azureedge.net/dotnet/Sdk/3.0.100-preview-009817/dotnet-sdk-3.0.100-preview-009817-osx-x64.tar.gz
+
+download-macos:
+	curl $(URL) --output dotnet-sdk-3.tar.gz
+	tar -xvf dotnet-sdk-3.tar.gz
+
+build-sample:
+	./dotnet build HelloWorld
+
+SHAREDRUNTIME:=shared/Microsoft.NETCore.App/3.0.0-preview-27122-01/
```

Comment:
ah ok, this is the bit I was missing - you just drop overwrite the normal coreCLR runtime with mono and trick `./dotnet` into running it.  Good enough for hacking.